### ISSUE TITLE
Project creation and storage

### ISSUE THREAD

**Issue by robinp7720** [Newawe/newawe#7]
There are a few things that will decide the way I will implement users and project creation and storage. The first is weather or not this will primary be a git client (aka github/bitbucket) or an online code editor with free storage (Scratch style). 

If we're going to go the git route, we will need to create a php based git client (completaly remake git in php).

Making this scratch style would be much easier but limit you with the web based editor. You would also be able to upload files so you can use your own editor.

If we're going git style there would also be some disadvantages. Git is made with sync development in mind. Using scratch style we could implement something like the google docs live editor.

Git could always be implemented after if we want to.

**Comment by skistaddy** [comment]
Let's start with the 'Scratch Style'

**Comment by ghost** [comment]
I know where to find scratch style. The scratch wiki skin. It will contain everything we need.

**Comment by robinp7720** [comment]
Not what I was talking about @Radical-from-Scratch

**Comment by jeandrek** [comment]
Clone vi and emacs and a 'simple' editor in the browser, and allow easy downloading/uploading for people who want to use their own editor?

**Comment by skistaddy** [comment]
@Jonathan50 that would be good!

**Comment by skistaddy** [comment]
@Jonathan50, would you be able to help with the create page? You seem to know what to do.

**Comment by as-com** [comment]
[Ace Editor](https://ace.c9.io/#nav=about) is a good option.

**Comment by skistaddy** [comment]
@as-com Do you know how to add it in?

**Comment by as-com** [comment]
@bjskistad First, we need to get an actually functioning backend before we can consider an interface for uploading/creating projects.

**Comment by skistaddy** [comment]
@as-com I believe @robinp7720 was doing that. But he's not on currently (He lives in a different timezone), would you be able to read his code and see what he was trying to do?

**Comment by jeandrek** [comment]
> @Jonathan50, would you be able to help with the create page? You seem to know what to do.

@robinp7720 can 'views' have PHP code? Or what do you do instead?

**Comment by ghost** [comment]
Oh wait, assigning me seemed to have unassigned you. That's weird :/

**Comment by ghost** [comment]
Accidental lock, reopening

**Comment by ghost** [comment]
I think it should be like Scratch, only with code and not projects.

**Comment by skistaddy** [comment]
@Radical-from-Scratch I agree

**Comment by ghost** [comment]
Also, let's make a "Newawe" git group, I will do that now.

**Comment by skistaddy** [comment]
There's already a user.

**Comment by skistaddy** [comment]
And I'll make it. :)

**Comment by ghost** [comment]
Okay. Also, was the "There's already a user" comment intended for me?

**Comment by skistaddy** [comment]
Yeah. Sorry. XD

**Comment by ghost** [comment]
I did not understand the meaning. Could you reword it.

**Comment by skistaddy** [comment]
Someone already has that username. invited you to the 'newaweteam'.

**Comment by ghost** [comment]
Well I didn't mean the group would be called "Newawe" I just meant we would have a group for Newawe

**Comment by jeandrek** [comment]
Uh, why are there two orgs? @newawe and @newaweteam.
The repo is owned by @newawe. And I've been invited to both...

**Comment by skistaddy** [comment]
Sorry. Didn't realize I owned @newawe XD

**Comment by jeandrek** [comment]
I have requested to be added to the @newawe/developers team.

**Comment by skistaddy** [comment]
K!

**Comment by jeandrek** [comment]
@as-com 

> [Ace Editor](https://ace.c9.io/#nav=about) is a good option.

Cool. Can it emulate vi/emacs?

**Comment by jeandrek** [comment]
Tosh uses [CodeMirror](https://codemirror.net/).

Also, what about a free/libre license? Not a big deal if users don't get benefits from modifying Newawe. (Not that this issue is the right place to discuss licensing... Should I make a new issue?)

**Comment by as-com** [comment]
@Jonathan50 [Yes](https://ace.c9.io/build/kitchen-sink.html)

**Comment by jeandrek** [comment]
> @Jonathan50 [Yes](https://ace.c9.io/build/kitchen-sink.html)

Is it just my browser or is it really, really buggy?
Spaces and tabs don't work and mess up the cursor.

**Comment by as-com** [comment]
@Jonathan50 It's probably your browser. What are you using?

**Comment by jeandrek** [comment]
Firefox Nightly. Not really the most stable browser :package: 

> @Jonathan50, would you be able to help with the create page? You seem to know what to do.

I'm going to try on the branch `Jonathan50-attempt-to-make-create-page`. Don't expect anything to come from it though.

**Comment by jeandrek** [comment]
I'll try in Iceweasel. It's a lot more stable.

**Comment by jeandrek** [comment]
It works fine in Iceweasel.

**Comment by jeandrek** [comment]
Here's what I've done so far:
![my attempt at making the Create page](http://i.cubeupload.com/Y1KLJA.png)
Is this okay?

**Comment by jeandrek** [comment]
I'm using JS in `<script>` tags. Is that a bad idea? Where should I put a JS file?

I think the function definitions should go in a seperate file but the

``` js
<script>
var thisProject = new Project(prompt("Name?"));
</script>
```

should stay.

**Comment by jeandrek** [comment]
For now I've put it in `asssets/js/project.js`.

**Comment by ghost** [comment]
Okay, this website is CLEARLY only intended for people with js enabled. I'll set up something for people that have it off.

**Comment by robinp7720** [comment]
@Jonathan50 No, php will not be executed in the views. You can pass HTML through the variables though. I might  also implement if and while loops and arrays into the view manger.

Or we could switch to something like twig?

**Comment by jeandrek** [comment]
@robinp7720 How would you do something like... make a page which a form should send a pull request to? It should probably be done in PHP.

Oh you could just make a new PHP page in the root. You wouldn't need to use the templating/views thing because there isn't any boilerplate PHP or HTML and users aren't supposed to see the page. Right?

**Comment by jeandrek** [comment]
@Radical-from-Scratch 

> Oh wait, assigning me seemed to have unassigned you. That's weird :/

Yes, only one user can be assigned at a time :confused:

**Comment by ghost** [comment]
At first I was like, "Yay! You can assign multiple people now!" and then I remembered I don't want to do this.
